backend/src/controllers/MyRestaurantController.ts

```diff
@@ -22,21 +27,19 @@ const createMyRestaurant = async (req: Request, res: Response) => {
     await restaurant.save();
     res.status(201).send(restaurant);
   } catch (error) {
-    console.log(error);
-    res.status(500).json({ message: 'Error creating restaurant' });
+    next(error); // Pass error to the centralized handler
   }
 };
 
-const getMyRestaurant = async (req: Request, res: Response) => {
+const getMyRestaurant = async (req: Request, res: Response, next: NextFunction) => {
   try {
     const restaurant = await Restaurant.findOne({ user: req.userId });
     if (!restaurant) {
-      return res.status(404).json({ message: 'Restaurant not found' });
+      throw new AppError('Restaurant not found', 404);
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/omkardongre/Food-Ordering-App/pull/47#issuecomment-2430019945>#d33549</a><span> - 10/22/2024, 07:15 pm</span>
<br>

<div><small>🟠</small> Medium importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> Consider using a more specific error message that provides context about why the restaurant was not found. This could help with debugging and improve the API's usability.
</div>
<div id="fix">
<b>Fix:</b> Provide a more detailed error message that includes context about the user ID or any other relevant information.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -38 +38 @@
 -      throw new AppError('Restaurant not found', 404);
 +      throw new AppError(`Restaurant not found for user ID: ${req.userId}`, 404);
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged